### ISSUE TITLE
Move C++ websocket server code back to ws-protocol repo

### ISSUE THREAD

**Issue by jtbandes** [foxglove/ros-foxglove-bridge#89]
Fix [https://github.com/foxglove/ros-foxglove-bridge/issues/89](https://github.com/foxglove/ros-foxglove-bridge/issues/89)

@jacob said:

> [https://github.com/foxglove/ros-foxglove-bridge/issues/89#issuecomment-2343714056](https://github.com/foxglove/ros-foxglove-bridge/issues/89#issuecomment-2343714056)
>
> this sounds nice, but am I remembering correctly that the ros folks were against FetchContent in the build process? or did we find some agreeable workaround?
>
> [Comment on #89 Move C++ websocket server code back to ws-protocol repo](https://github.com/foxglove/ros-foxglove-bridge/issues/89#issuecomment-2343714056)

**Comment by jhurliman** [comment]
I suggest moving this issue to https://github.com/foxglove/ws-protocol since the result would be a PR to that repo and no change to this repo.

**Comment by jtbandes** [comment]
I'd rather not have two copies of the code and have to keep them in sync, so the result should also include deleting the version in this repo.

**Comment by amacneil** [comment]
It sounds like time for a monorepo

![image](https://user-images.githubusercontent.com/637671/208624472-d1dcb13f-e882-43fb-af5d-6ed96639b9a2.png)

**Comment by jlack1987** [comment]
Which implementation of the websocket protocol is most up-to-date/newer? It seems [this one](https://github.com/foxglove/ws-protocol) has more recent commits than this repo no?

**Comment by achim-k** [comment]
> Which implementation of the websocket protocol is most up-to-date/newer? It seems [this one](https://github.com/foxglove/ws-protocol) has more recent commits than this repo no?

This repo basically uses a copy of the C++ server code in https://github.com/foxglove/ws-protocol, with only a few minor differences.

I'm not sure if there is an elegant way to add `ws-protocol` as a build dependency remove the server code in this pacakge. Maybe we could do so using [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html)?

**Comment by jlack1987** [comment]
Yeah it's doable via the `FetchContent` stuff in cmake. I actually did just that on that repo a few days ago. That's when I ran into this repo as well and got confused. Some minor cmake mods are needed though on `ws-protocol` to ensure some of `websocketpp`'s dependencies are available(zlib, openssl, and boost iirc). Very easy stuff to do though

**Comment by amacneil** [comment]
We're currently working on a new C++ SDK, after which we will update `ros-foxglove-bridge` to use the new SDK, and figure out shared libraries at that time.
